### DISCUSSION TITLE
Adding a date option as an input type.

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#128] [Help] [answered]
Hi,

We're trying to use Swell to see virtual events as products and everything else seems fine except I'm having a hard time figuring out how we can add a date to a product.  Eg, when the user is on a product page one of the options needs to be an option for the date of an event.  We can probably do with just using a text field for now as many people don't actually have an exact date when they are booking but having the option of a date input type on a product would be ideal.

Is there the possibility of adding this in the future or perhaps I am missing another work around? eg should I somehow instead update the model for a cart item with this field?

Any suggestions would be greatly apprecated.

Thanks

<img width="796" alt="CleanShot 2023-02-03 at 14 55 52@2x" src="https://user-images.githubusercontent.com/48241/216621201-1d648f73-b3f9-4f5a-b6fd-6c249f00377b.png">

**Comment by ericingram** [comment]
You could implement a date picker on the front-end and store the value in a custom product option, in the cart or elsewhere depending on what you're building.

That said, we will likely improve this in the future by introducing the same content field types you see in the Model editor:

<img width="786" alt="image" src="https://user-images.githubusercontent.com/952134/216650193-a15be836-7d04-4a6e-af89-6d558bd4d60e.png">

**Comment by holden** [comment]
Wow, that looks perfect.  When can we get that?

**Reply by ericingram** [reply, marked as answer]
It's part of the content models, which allow you to add custom fields that can be edited in the admin dashboard for managing custom content generally. [See docs here](https://developers.swell.is/guides/data-model-customization#content-models).

However, I was pointing out that we might eventually add these field types as product option inputs also, but for now you still have to do it in the frontend yourself.
